### PR TITLE
[Test] downgrade nbgitpuller and jupyterhub in dev-r hub image

### PR DESCRIPTION
**Pull request by balajialg**
@shaneknapp Let me know if you are okay with making changes to infra-req for dev-r? I will test this in staging and then revert the changes. Yet another data point that we can use to report upstream.

### CODE REVIEW

**Comment by shaneknapp** [comment]
this won't work as expected due to the hub image itself having a higher version of jupyterhub installed.  you'll need to build and push a new image w/the lower deps installed.

i believe that you can specify this in `staging.yml` as `hub.image.<etc>`.  @ryanlovett have you done this before?

**Review by shaneknapp — COMMENTED** [review]
this won't work as expected due to the hub image itself having a higher version of jupyterhub installed.  you'll need to build and push a new image w/the lower deps installed.

i believe that you can specify this in `staging.yml` as `hub.image.<etc>`.  @ryanlovett have you done this before?

**Comment by shaneknapp** [comment]
i'd also considering experimenting by also lowering the jupyterhub version to 4.0.x and keeping nbgitpuller at 1.2.1

**Comment by balajialg** [comment]
@shaneknapp Got it, thanks. If you can redirect me to a documentation then I will take a look at it. Or else we can probably do this during a short check-in if you are available.

**Comment by ryanlovett** [comment]
@shaneknapp I haven't done this exact thing before, but it should be possible to set a value in a deployment that masks any corresponding value in hub/values.yml. This would indeed be hub.image. To build a new hub image you'd modify images/hub/Dockerfile and build a new hub image with chartpress.

chartpress builds and pushes a new image, but it also changes the hub version in the top-level config automatically. So either you would let it do that, test a hub with hubploy, and not commit those changes until you want to deploy them for all hubs; or you would remove the top-level config changes to hub.image and manually insert them into the testing hub config, and commit that.

**Comment by shaneknapp** [comment]
> @shaneknapp I haven't done this exact thing before, but it should be possible to set a value in a deployment that masks any corresponding value in hub/values.yml. This would indeed be hub.image. To build a new hub image you'd modify images/hub/Dockerfile and build a new hub image with chartpress.
> 
> chartpress builds and pushes a new image, but it also changes the hub version in the top-level config automatically. So either you would let it do that, test a hub with hubploy, and not commit those changes until you want to deploy them for all hubs; or you would remove the top-level config changes to hub.image and manually insert them into the testing hub config, and commit that.

ah i see.  i created a PR to do that...  could you take a look and we can carry that convo on over there?  https://github.com/berkeley-dsep-infra/datahub/pull/5696

**Comment by shaneknapp** [comment]
> > @shaneknapp I haven't done this exact thing before, but it should be possible to set a value in a deployment that masks any corresponding value in hub/values.yml. This would indeed be hub.image. To build a new hub image you'd modify images/hub/Dockerfile and build a new hub image with chartpress.
> > chartpress builds and pushes a new image, but it also changes the hub version in the top-level config automatically. So either you would let it do that, test a hub with hubploy, and not commit those changes until you want to deploy them for all hubs; or you would remove the top-level config changes to hub.image and manually insert them into the testing hub config, and commit that.
> 
> ah i see. i created a PR to do that... could you take a look and we can carry that convo on over there? #5696

figured it out! https://github.com/berkeley-dsep-infra/datahub/pull/5696#issuecomment-2076026761

File: deployments/dev-r/images/default/infra-requirements.txt [hunk 1]
```diff
@@ -10,10 +10,10 @@
 # FIXME: Freeze this to get exact versions of all dependencies
 notebook==7.0.7
 jupyterlab==4.0.11
-nbgitpuller==1.2.1
+nbgitpuller==1.1.1
 jupyter-resource-usage==1.0.1
 # Matches version in images/hub/Dockerfile
-jupyterhub==4.1.5
+jupyterhub==4.0.1
```
Comment (shaneknapp) [inline deployments/dev-r/images/default/infra-requirements.txt]: change this to `4.0.2`, which matches the version on the dev-r staging hub.

**Review by shaneknapp — CHANGES_REQUESTED** [review]

File: deployments/dev-r/images/default/infra-requirements.txt (hunk 1 above)
Comment (balajialg) [reply deployments/dev-r/images/default/infra-requirements.txt]: done

**Review by shaneknapp — APPROVED** [review]

**Comment by shaneknapp** [comment]
i would suggest that after merging this and testing, to open another PR to bump nbgitpuller to `1.2.1` .

**Comment by balajialg** [comment]
@shaneknapp Sounds good. Will do

**Pull request merged**